### DISCUSSION TITLE
Attach keyboard to html chat

### DISCUSSION THREAD

**Discussion by christogonus** [def-studio/telegraph#460] [Q&A]
In  a custom webhook response to command /start, I want to have a keyboard on the chat, like below

`$this->chat->html($message)
            ->keyboard(Keyboard::make()->buttons([
                Button::make('Delete')->action('delete')->param('id', '42'),
                Button::make('open')->url('https://test.it'),
                Button::make('Web App')->webApp('https://web-app.test.it'),
                Button::make('Login Url')->loginUrl('https://loginUrl.test.it'),
            ]))->send();`

I have imported the classes as well, 

`use DefStudio\Telegraph\Keyboard\Button;
use DefStudio\Telegraph\Keyboard\Keyboard;`

But I get response as below

`[2023-11-01 06:25:17] local.ERROR: Class "App\Services\Keyboard" not found {"exception":"[object] (Error(code: 0): Class \"App\\Services\\Keyboard\" not found at /home/spatryoc/tg.spatryoc.net/app/Services/CustomWebhookHandler.php:45)`

I tried the following:

1. Inline the class namespace
2. run the command "composer dump-autoload"

but none of those helped me get desired result, which is to post html content with keyboard actions.
When I remove the ->keyboard() part, the chat responds to command.

How can I get this work with ->keyboard()?

**Comment by dostonadilov** [comment]
Same problem

**Comment by fabio-ivona** [comment]
This code is working:

```php
 Telegraph::html('test message')
        ->keyboard(
            \DefStudio\Telegraph\Keyboard\Keyboard::make()->buttons([
                \DefStudio\Telegraph\Keyboard\Button::make('Delete')->action('delete')->param('id', 42),
                \DefStudio\Telegraph\Keyboard\Button::make('Open')->url('https://test.it'),
                \DefStudio\Telegraph\Keyboard\Button::make('Login Url')->loginUrl('https://loginUrl.test.it'),
            ])
        )
        ->send()->dump();
```

the `dump()` call should allow you to have more info about the call result

**warn** your button's URLs should be valid URLs, otherwise you'll obtain a 400 BOT_DOMAIN_INVALID error
